PatternPal/PatternPal.LoggingServer/Services/LogJobs.cs

```diff
@@ -0,0 +1,38 @@
+
+using PatternPal.LoggingServer.Data;
+using Quartz;
+
+namespace PatternPal.LoggingServer.LogJobs;
+/// <summary>
+/// Job that clears the codestates table every Monday at midnight
+/// </summary>
+public class ClearCodestatesJob : IJob
+{
+    private readonly ILogger<ClearCodestatesJob> _logger;
+    private readonly EventRepository _eventRepository;
+
+    public ClearCodestatesJob(ILogger<ClearCodestatesJob> logger, EventRepository repository)
+    {
+        _logger = logger;
+        _eventRepository = repository;
+    }
+    
+    public async Task Execute(IJobExecutionContext context)
+    {
+
+        _logger.LogInformation("Clearing codestates table");
+        List<Guid> uniqueCodeStates = await _eventRepository.GetUniqueCodeStates();
+
+        foreach (string folderName in Directory.EnumerateDirectories("codestates"))
```

Comment:
Maybe it's good practice to define a constant for the CodeStates-path. Also, I know folders are not capital-matched, but if the folder is called `CodeStates` we shouldn't be looking for `codestates` here.